logic/atrium-logic-common/src/main/kotlin/ch/tutteli/atrium/logic/impl/DefaultIterableLikeAssertions.kt

```diff
@@ -136,4 +136,45 @@ class DefaultIterableLikeAssertions : IterableLikeAssertions {
             }
             .withoutOptions()
             .build()
+
+    override fun <T : Any, E> containsNoDuplicates(
+        container: AssertionContainer<T>,
+        converter: (T) -> Iterable<E>
+    ): Assertion {
+        val iterableContainer = container.changeSubject.unreported(converter).toAssertionContainer()
+        return LazyThreadUnsafeAssertionGroup {
+            val list = turnSubjectToList(iterableContainer).maybeSubject.getOrElse { emptyList() }
+            val hasElementAssertion = createHasElementAssertion(list.iterator())
+
+            val duplicates: List<Pair<Int, E>> = list
+                .mapWithIndex()
+                .filter { (_, element) ->
+                    list.count { e -> e == element } > 1
+                }
+                .map { (index, element) -> index to element }
```

Comment:
Is this required?

Reply:
check duplicates and then map from WithIndex class to Pair....

Reply:
remove useless line of code